pkg/api/azenvtypes.go

```diff
@@ -7,18 +7,18 @@ import "fmt"
 
 //AzureEnvironmentSpecConfig is the overall configuration differences in different cloud environments.
 type AzureEnvironmentSpecConfig struct {
-	CloudName            string
-	DockerSpecConfig     DockerSpecConfig
-	KubernetesSpecConfig KubernetesSpecConfig
-	DCOSSpecConfig       DCOSSpecConfig
-	EndpointConfig       AzureEndpointConfig
-	OSImageConfig        map[Distro]AzureOSImageConfig
+	CloudName            string                        `json:"cloudName,omitempty"`
+	DockerSpecConfig     DockerSpecConfig              `json:"dockerSpecConfig,omitempty"`
+	KubernetesSpecConfig KubernetesSpecConfig          `json:"kubernetesSpecConfig,omitempty"`
+	DCOSSpecConfig       DCOSSpecConfig                `json:"-"`
```

Comment:
Is `json:"-"` preferable to simply leaving out the struct tag altogether?

Reply:
yes,  `json:"-"` , DCOSSpecConfig will not show up in the JSON output during Marshal, 
The field is still show up in JSON output even it leave out the struct tag.